File: src/js/bruker2dicom/modules/patient.js
require('bruker2dicom/common.js');

var _module = namespace('dicomifier.bruker2dicom.modules');

_module.Patient = function(indexGenerator, dicomDataset, brukerDataset) {
    var toDicom = dicomifier.bruker2dicom.toDicom;
    var dictionaryMapper = dicomifier.bruker2dicom.dictionaryMapper;

    toDicom(
        indexGenerator, dicomDataset, 'PatientName',
        brukerDataset, 'VisuSubjectName', 2);

    toDicom(
        indexGenerator, dicomDataset, 'PatientID',
        brukerDataset, 'VisuSubjectId', 2);

    toDicom(
        indexGenerator, dicomDataset, 'PatientBirthDate',
        brukerDataset, 'VisuSubjectBirthDate', 2);

    toDicom(
        indexGenerator, dicomDataset, 'PatientSex',
        brukerDataset, 'VisuSubjectSex', 2,
        dictionaryMapper({
            'MALE': 'M', 'FEMALE': 'F', 'UNDEFINED': 'O', 'UNKNOWN': 'O' }));

    toDicom(
        indexGenerator, dicomDataset, 'PatientComments',
        brukerDataset, 'VisuSubjectComment', 3);

    dicomDataset[dicomifier.dictionary['PatientSpeciesDescription'][1]] = {
        'vr': dicomifier.dictionary['PatientSpeciesDescription'][0],
        'Value' : ['UNKNOWN'] };

    dicomDataset[dicomifier.dictionary['PatientBreedDescription'][1]] = {
        'vr': dicomifier.dictionary['PatientBreedDescription'][0] };

    dicomDataset[dicomifier.dictionary['PatientBreedCodeSequence'][1]] = {
        'vr': dicomifier.dictionary['PatientBreedCodeSequence'][0] };

    dicomDataset[dicomifier.dictionary['BreedRegistrationSequence'][1]] = {
        'vr': dicomifier.dictionary['BreedRegistrationSequence'][0] };

    dicomDataset[dicomifier.dictionary['ResponsiblePerson'][1]] = {
        'vr': dicomifier.dictionary['ResponsiblePerson'][0] };

    dicomDataset[dicomifier.dictionary['ResponsibleOrganization'][1]] = {
        'vr': dicomifier.dictionary['ResponsibleOrganization'][0] };
};

File: src/python/dicomifier/bruker_to_dicom/modules/patient.py
#########################################################################
# Dicomifier - Copyright (C) Universite de Strasbourg
# Distributed under the terms of the CeCILL-B license, as published by
# the CEA-CNRS-INRIA. Refer to the LICENSE file or to
# http://www.cecill.info/licences/Licence_CeCILL-B_V1-en.html
# for details.
#########################################################################

from . import cached

Patient= [ # PS 3.3, C.7.1.1
    ("VisuSubjectName", "PatientName", 2, None),
    ("VisuSubjectId", "PatientID", 2, None),
    ("VisuSubjectBirthDate", "PatientBirthDate", 2, None),
    (
        "VisuSubjectSex", "PatientSex", 2,
        cached("__PatientSex")(
            lambda d,g,i: [{
                    "MALE": "M", "FEMALE": "F", "UNDEFINED": "O", "UNKNOWN": "O",
                    None: None
                }[d["VisuSubjectSex"][0]]])),
    ("VisuSubjectComment", "PatientComments", 3, None),
    (None, "PatientBreedDescription", 2, lambda d,g,i: None),
    (None, "PatientBreedCodeSequence", 2, lambda d,g,i: None),
    (None, "BreedRegistrationSequence", 2, lambda d,g,i: None),
    (None, "ResponsiblePerson", 2, lambda d,g,i: None),
    (None, "ResponsibleOrganization", 2, lambda d,g,i: None)
]
